src/app/dataset-view/dataset.component.ts

```diff
@@ -63,14 +63,20 @@ export class DatasetComponent
         this.checkWindowSize();
         this.trackSubscription(
             this.router.events
-                .pipe(filter((event) => event instanceof NavigationEnd))
+                .pipe(
+                    filter((event) => event instanceof NavigationEnd),
+                    tap(() => this.getDatasetDataByLineageView()),
```

Comment:
Why something special for lineage view?
I'm expecting to see reaction on dataset ID change - if it has not changed, there is nothing to request. If it has, you need to request full data.